integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/TopicBasicsTestCase.java

```diff
@@ -0,0 +1,162 @@
+
+    @BeforeClass
+    public void init() {
+        super.init();
+    }
+
+    /**
+     * In this test method we check sending messages to a Topic and receiving them.
+     * 1. Register two subscribers to a Topic.
+     * 2. Send a message.
+     * 3. Receive them by two subscribers.
+     * 4. Send 500 messages.
+     * 5. Receive them by two subscribers.
+     * @throws JMSException
+     * @throws NamingException
+     * @throws IOException
+     */
+    @Test(groups = "wso2.mb.topic")
+    public void performTopicSendReceive() throws JMSException, NamingException, IOException,
+            InterruptedException {
+
+        TopicSender sender = null;
+        TopicReceiver subscriber = null;
+        TopicReceiver subscriber2 = null;
+        try {
+            subscriber = new TopicReceiver("testTopic", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    null);
+
+            subscriber2 = new TopicReceiver("testTopic", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    null);
+
+            //send a message
+            sender = new TopicSender("testTopic", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader);
+            sender.sendMessages(1, "test message");
+
+            Assert.assertNotNull(subscriber.receiveMessage(1000L), "Topic not received by the subscriber");
+            Assert.assertNotNull(subscriber2.receiveMessage(1000L), "Topic not received by the 2nd subscriber");
+
+            //send 500 messages
+            subscriber.registerMessageListener();
+            subscriber2.registerMessageListener();
+            sender.sendMessages(500, "test message");
+            TimeUnit.SECONDS.sleep(5L);
+            Assert.assertEquals(subscriber.receivedMessageCount(), 500, "Did not receive all the topics");
+            Assert.assertEquals(subscriber2.receivedMessageCount(), 500, "Did not receive all the topics");
+        } finally {
+            if (sender != null) {
+                sender.closeSender();
+                subscriber.closeReceiver();
+            }
+        }
+    }
+
+    /**
+     * Test for durable subscription.
+     * 1. Create two durable subscribers.
+     * 2. Send 10 messages and check for their reception.
+     * 3. Close one subscriber.
+     * 4. Send 10 messages.
+     * 5. Start the closed subscriber again.
+     * 6. Check the reception of messages.
+     * @throws JMSException
+     * @throws NamingException
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    @Test(groups = "wso2.mb.topic")
+    public void testDurableSubscription() throws JMSException, NamingException, IOException, InterruptedException {
+        TopicSender sender = null;
+        TopicReceiver subscriber = null;
+        TopicReceiver subscriber2 = null;
+        try {
+            subscriber = new TopicReceiver("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    "sub1");
```

Comment:
Can we merge this into the previous line?

Reply:
Then we can't close the subscribers inside finally block.